server/inc/server.h

```diff
@@ -15,6 +17,7 @@ typedef struct s_client {
     GDataInputStream *data_in;
     GDataOutputStream *data_out;
     gint64 uid;
+    void (*request_handler[RQ_RECOVERY_PASSWD + 1])();
```

Comment:
У чому сенс додавання одиниці до константи? Й взагалі я не бачу сенсу його тримати в структурі клієнту

Reply:
Це виходить, що для кожного клієнта, що підключиться до нас ми буде створювати масив веазівників на функції?

Reply:
we have already discussed about this one and u understood that u can't transfer handler as argument into callback func

Reply:
> У чому сенс додавання одиниці до константи? Й взагалі я не бачу сенсу його тримати в структурі клієнту

Because everything starts counting from 0 but in case when we have size we count objects inserting to array so when i put size = 4 and then i try to get array[4] i get segfault.

Reply:
So?...

Reply:
I do not understand the problem

> > У чому сенс додавання одиниці до константи? Й взагалі я не бачу сенсу його тримати в структурі клієнту
> 
> Because everything starts counting from 0 but in case when we have size we count objects inserting to array so when i put size = 4 and then i try to get array[4] i get segfault.

We cannot access array starting from 0?

Reply:
> we have already discussed about this one and u understood that u can't transfer handler as argument into callback func

Yes, and I said that is bad idea

Reply:
Advise you to use as last element SIZE, so we won't need to change this part of code